biogrid/libs/biogrid-simulator/src/lib/biobattery/biobattery-class.ts

```diff
@@ -1,18 +1,45 @@
-import { Battery, Energy, Validatable, validate } from "@biogrid/grid-simulator";
+import {
+  Battery,
+  Distance,
+  Energy,
+  ItemPosition,
+  Validatable,
+  validate,
+} from '@biogrid/grid-simulator';
 
 export class BioBattery implements Battery {
   private currentBatteryPower: Energy;
   private readonly maxCapacity: Energy;
+  private readonly position: ItemPosition;
 
-  constructor(currentBatteryPower: Energy = 0, maxCapacity: Energy = 10) {
+  /**
+   * Take in a position which is generated by the grid class so that batteries
+   * are evenly spaced out.
+   * The positions have been verified in the Biogrid class
+   * @param x Distance from the left side
+   * @param y Distance from the top
```

Comment:
nit: left side/top of what? left edge of town constraint and top edge of town constraint?